### DISCUSSION TITLE
Can quarto work with Jekyll? or allow for a fully custom front-end?

### DISCUSSION THREAD

**Discussion by lwasser** [quarto-dev/quarto-cli#6534] [Q&A]
### Description

Hi - i have a general question about how flexible quarto is. Im developing a [tutorial website ](https://datascienceskills.org/) that will host open lessons as a part of the [pyOpenSci effort](https://www.pyopensci.org). I am not using quarto as a back end but i'm fighting some with trying to override styles and templates to create my own design but still allow quarto to build pages and produce code output. 

i'm very familiar with customizing jekyll sites, creating templates, writing liquid, etc. but am not sure if i can do that easily with quarto? Is there a way for me to produce custom html templates, SASS css files and includes that i can use to style and design the site (possibly using jekyll?) while still using quarto as the back end to build the pages / run code?

thank you.

**Comment by rgaiacs** [comment]
Dear @lwasser,

Quarto has support to [Hugo](https://gohugo.io/), see https://quarto.org/docs/output-formats/hugo.html, that is similar to Jekyll.

Based on my experience, I do **not** recommend using Quarto to render Jupyter Notebooks into Markdown and pass the Markdown to Jekyll because of

1. Markdown output is not 100% supported, see https://github.com/quarto-dev/quarto-cli/issues/5830.
2. Some of Quarto features, for example [Callout Blocks](https://quarto.org/docs/authoring/callouts.html), are extensions to Pandoc Markdown and will have to be re-implemented in Jekyll
3. Quarto uses Pandoc and Jekyll uses another Markdown parser and minor differences can create some issues that are time consuming to trace back

Quarto has a great support to style customisation, see https://quarto.org/docs/presentations/revealjs/themes.html#customizing-themes, where you can include your own CSS.

**Comment by mcanouil** [comment]
An example of a Quarto extension providing a complete HTML template to be used for website and single HTML documents: <https://github.com/mccarthy-m-g/quarto-lcars-theme>.

You can see some other website customisation examples on the Awesome Quarto list: <https://github.com/mcanouil/awesome-quarto#websites-formats>.
